container/registry.py

```diff
@@ -169,6 +169,7 @@ def _image_exists(image_reference: str) -> bool:
   return False
 
 
+@functools.lru_cache
```

Comment:
hmmmmm....

tbh, I think this is a bad idea. it may be a constructed corner case. but what if the digest _does_ change while we still hold the old, cached value? in that case we would miss this.

Reply:
Fair enough. I don't think that we have many instances of this coding running long enough for this to be a likely occurrence, but I figure that it'd be a real head-scratcher if it _does_ happen.
I'll remove the cache.

Reply:
you never know what functionality we shall mv to `whd` in the future. which is a long-running instance